### PR TITLE
Detach integration point when RootNode leaves composition

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Fixes https://github.com/bumble-tech/appyx/issues/20.

Detach integration point when RootNode leaves composition

## Check list

- [ ] I have updated `CHANGELOG.md` if required.
- [ ] I have updated documentation if required.

### CODE REVIEW

File: core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt [hunk 1]
```diff
@@ -50,7 +50,10 @@ fun <N : Node> NodeHost(
             node.updateLifecycleState(source.lifecycle.currentState)
         }
         lifecycle.addObserver(observer)
-        onDispose { lifecycle.removeObserver(observer) }
+        onDispose {
+            lifecycle.removeObserver(observer)
+            integrationPoint.detach()
```
Comment (LachlanMcKee) [inline core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt:55]: Might be wise to add a test for this

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
  - Migrate [app-tree-utils](https://github.com/badoo/app-tree-utils) into this repository.
  - Do not allow setting `Node.integrationPoint` on non-root nodes.
+ - Fix integration point attached twice crash
```
[the remark this answers is not in the document]
Comment (LachlanMcKee) [reply CHANGELOG.md:7]: Not specifically for this PR, but in general should we link to the Github issue/PR in the changelog?

[the remark this answers is not in the document]
Comment (KovalevAndrey) [reply CHANGELOG.md:7]: yeah, probably we should. I'll add a link to the issue

**Review by CherryPerry — APPROVED** [review]

File: core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt (hunk 1 above)
Comment (KovalevAndrey) [reply core/src/main/java/com/bumble/appyx/core/integration/NodeHost.kt]: can't unit test, leaving as is for now
